From 651248610ab7a5a3b340bd18b80fc53ae692d8b4 Mon Sep 17 00:00:00 2001
From: Peter Vary <pvary@cloudera.com>
Date: Fri, 23 Feb 2018 09:08:32 +0100
Subject: [PATCH 1386/1431] CDH-65073: HIVE-18771: Refactor tests, so only 1
 MetaStore instance will be started per test class
 and test configuration (Peter Vary, reviewed by
 Sahil Takiar)

(cherry picked from commit f9768af0e147539c52311643c8bcaa778483e051)

Change-Id: I2ddc7f8ea5cc65e7bacfd820f39ee80ba27c7fb7
(cherry picked from commit d19b7dd2ea5e634989332fca4fbb48c55d80e985)
(cherry picked from commit 8c35c426cbd3c3ba994a3e0642173b0f109a43ed)
---
 .../hive/metastore/client/MetaStoreClientTest.java |   93 ++++++++++++++++++++
 .../metastore/client/TestAddAlterDropIndexes.java  |   31 +------
 .../hive/metastore/client/TestAddPartitions.java   |   30 +------
 .../client/TestAddPartitionsFromPartSpec.java      |   32 +------
 .../hive/metastore/client/TestAlterPartitions.java |   31 +------
 .../metastore/client/TestAppendPartitions.java     |   30 +------
 .../hive/metastore/client/TestDatabases.java       |   41 +--------
 .../hive/metastore/client/TestDropPartitions.java  |   36 ++------
 .../hive/metastore/client/TestFunctions.java       |   31 +------
 .../hive/metastore/client/TestGetListIndexes.java  |   31 +------
 .../hive/metastore/client/TestGetPartitions.java   |   33 +------
 .../hive/metastore/client/TestGetTableMeta.java    |   32 +------
 .../hive/metastore/client/TestListPartitions.java  |   34 +------
 .../client/TestTablesCreateDropAlterTruncate.java  |   36 ++------
 .../hive/metastore/client/TestTablesGetExists.java |   32 +------
 .../hive/metastore/client/TestTablesList.java      |   37 +-------
 16 files changed, 133 insertions(+), 457 deletions(-)
 create mode 100644 metastore/src/test/org/apache/hadoop/hive/metastore/client/MetaStoreClientTest.java

diff --git a/metastore/src/test/org/apache/hadoop/hive/metastore/client/MetaStoreClientTest.java b/metastore/src/test/org/apache/hadoop/hive/metastore/client/MetaStoreClientTest.java
new file mode 100644
index 0000000..4ab1ce9
--- /dev/null
+++ b/metastore/src/test/org/apache/hadoop/hive/metastore/client/MetaStoreClientTest.java
@@ -0,0 +1,93 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.hive.metastore.client;
+
+import org.apache.hadoop.hive.conf.HiveConf;
+import org.apache.hadoop.hive.metastore.minihms.AbstractMetaStoreService;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Tests abstract class for running MetaStoreClient API tests.
+ */
+@RunWith(Parameterized.class)
+public abstract class MetaStoreClientTest {
+  private static final Logger LOG = LoggerFactory.getLogger(TestDatabases.class);
+
+  // Needed until there is no junit release with @BeforeParam, @AfterParam (junit 4.13)
+  // https://github.com/junit-team/junit4/commit/1bf8438b65858565dbb64736bfe13aae9cfc1b5a
+  // Then we should remove our own copy
+  private static Set<AbstractMetaStoreService> metaStoreServices = null;
+
+  @Parameterized.Parameters(name = "{0}")
+  public static List<Object[]> getMetaStoreToTest() throws Exception {
+    metaStoreServices = new HashSet<AbstractMetaStoreService>();
+    List<Object[]> result = MetaStoreFactoryForTests.getMetaStores();
+    for(Object[] test: result) {
+      metaStoreServices.add((AbstractMetaStoreService)test[1]);
+    }
+    return result;
+  }
+
+  @BeforeClass
+  public static void startMetaStores() {
+    startMetaStores(new HashMap<HiveConf.ConfVars, String>(), new HashMap<String, String>());
+  }
+
+  /**
+   * Utility method, which can be used to start MetaStore instances with specific configurations
+   * different from the default.
+   * @param msConf Specific MetaStore configuration values
+   * @param extraConf Specific other configuration values
+   */
+  static void startMetaStores(Map<HiveConf.ConfVars, String> msConf,
+                              Map<String, String> extraConf) {
+    for(AbstractMetaStoreService metaStoreService : metaStoreServices) {
+      try {
+        metaStoreService.start(msConf, extraConf);
+      } catch(Exception e) {
+        // Catch the exceptions, so every other metastore could be stopped as well
+        // Log it, so at least there is a slight possibility we find out about this :)
+        LOG.error("Error starting MetaStoreService", e);
+      }
+    }
+  }
+
+  @AfterClass
+  public static void stopMetaStores() {
+    for(AbstractMetaStoreService metaStoreService : metaStoreServices) {
+      try {
+        metaStoreService.stop();
+      } catch(Exception e) {
+        // Catch the exceptions, so every other metastore could be stopped as well
+        // Log it, so at least there is a slight possibility we find out about this :)
+        LOG.error("Error stopping MetaStoreService", e);
+      }
+    }
+  }
+}
diff --git a/metastore/src/test/org/apache/hadoop/hive/metastore/client/TestAddAlterDropIndexes.java b/metastore/src/test/org/apache/hadoop/hive/metastore/client/TestAddAlterDropIndexes.java
index 62ec4c3..46bf232 100644
--- a/metastore/src/test/org/apache/hadoop/hive/metastore/client/TestAddAlterDropIndexes.java
+++ b/metastore/src/test/org/apache/hadoop/hive/metastore/client/TestAddAlterDropIndexes.java
@@ -21,7 +21,6 @@
 import java.util.HashSet;
 import java.util.List;
 import java.util.Map;
-import java.util.Set;
 
 import org.apache.hadoop.fs.Path;
 import org.apache.hadoop.hive.metastore.IMetaStoreClient;
@@ -43,7 +42,6 @@
 import org.apache.thrift.TException;
 import org.apache.thrift.transport.TTransportException;
 import org.junit.After;
-import org.junit.AfterClass;
 import org.junit.Assert;
 import org.junit.Before;
 import org.junit.Test;
@@ -54,11 +52,7 @@
  * Tests for creating, altering and dropping indexes.
  */
 @RunWith(Parameterized.class)
-public class TestAddAlterDropIndexes {
-  // Needed until there is no junit release with @BeforeParam, @AfterParam (junit 4.13)
-  // https://github.com/junit-team/junit4/commit/1bf8438b65858565dbb64736bfe13aae9cfc1b5a
-  // Then we should remove our own copy
-  private static Set<AbstractMetaStoreService> metaStoreServices = null;
+public class TestAddAlterDropIndexes extends MetaStoreClientTest {
   private AbstractMetaStoreService metaStore;
   private IMetaStoreClient client;
 
@@ -68,29 +62,8 @@
   private static final String INDEX_TABLE_NAME = TABLE_NAME + "__" + INDEX_NAME + "__";
   private static final short MAX = -1;
 
-  @Parameterized.Parameters(name = "{0}")
-  public static List<Object[]> getMetaStoreToTest() throws Exception {
-    metaStoreServices = new HashSet<AbstractMetaStoreService>();
-    List<Object[]> result = MetaStoreFactoryForTests.getMetaStores();
-    for(Object[] test: result) {
-      metaStoreServices.add((AbstractMetaStoreService)test[1]);
-    }
-    return result;
-  }
-
-  public TestAddAlterDropIndexes(String name, AbstractMetaStoreService metaStore) throws Exception {
+  public TestAddAlterDropIndexes(String name, AbstractMetaStoreService metaStore) {
     this.metaStore = metaStore;
-    this.metaStore.start();
-  }
-
-  // Needed until there is no junit release with @BeforeParam, @AfterParam (junit 4.13)
-  // https://github.com/junit-team/junit4/commit/1bf8438b65858565dbb64736bfe13aae9cfc1b5a
-  // Then we should move this to @AfterParam
-  @AfterClass
-  public static void stopMetaStores() throws Exception {
-    for(AbstractMetaStoreService metaStoreService : metaStoreServices) {
-      metaStoreService.stop();
-    }
   }
 
   @Before
diff --git a/metastore/src/test/org/apache/hadoop/hive/metastore/client/TestAddPartitions.java b/metastore/src/test/org/apache/hadoop/hive/metastore/client/TestAddPartitions.java
index 0cb23e7..2c5b721 100644
--- a/metastore/src/test/org/apache/hadoop/hive/metastore/client/TestAddPartitions.java
+++ b/metastore/src/test/org/apache/hadoop/hive/metastore/client/TestAddPartitions.java
@@ -55,12 +55,7 @@
  * Tests for creating partitions.
  */
 @RunWith(Parameterized.class)
-public class TestAddPartitions {
-
-  // Needed until there is no junit release with @BeforeParam, @AfterParam (junit 4.13)
-  // https://github.com/junit-team/junit4/commit/1bf8438b65858565dbb64736bfe13aae9cfc1b5a
-  // Then we should remove our own copy
-  private static Set<AbstractMetaStoreService> metaStoreServices = null;
+public class TestAddPartitions extends MetaStoreClientTest {
   private AbstractMetaStoreService metaStore;
   private IMetaStoreClient client;
 
@@ -74,29 +69,8 @@
   private static final String MONTH_COL_NAME = "month";
   private static final short MAX = -1;
 
-  @Parameterized.Parameters(name = "{0}")
-  public static List<Object[]> getMetaStoreToTest() throws Exception {
-    metaStoreServices = new HashSet<AbstractMetaStoreService>();
-    List<Object[]> result = MetaStoreFactoryForTests.getMetaStores();
-    for(Object[] test: result) {
-      metaStoreServices.add((AbstractMetaStoreService)test[1]);
-    }
-    return result;
-  }
-
-  public TestAddPartitions(String name, AbstractMetaStoreService metaStore) throws Exception {
+  public TestAddPartitions(String name, AbstractMetaStoreService metaStore) {
     this.metaStore = metaStore;
-    this.metaStore.start();
-  }
-
-  // Needed until there is no junit release with @BeforeParam, @AfterParam (junit 4.13)
-  // https://github.com/junit-team/junit4/commit/1bf8438b65858565dbb64736bfe13aae9cfc1b5a
-  // Then we should move this to @AfterParam
-  @AfterClass
-  public static void stopMetaStores() throws Exception {
-    for(AbstractMetaStoreService metaStoreService : metaStoreServices) {
-      metaStoreService.stop();
-    }
   }
 
   @Before
diff --git a/metastore/src/test/org/apache/hadoop/hive/metastore/client/TestAddPartitionsFromPartSpec.java b/metastore/src/test/org/apache/hadoop/hive/metastore/client/TestAddPartitionsFromPartSpec.java
index 1b4dcec..fe80246 100644
--- a/metastore/src/test/org/apache/hadoop/hive/metastore/client/TestAddPartitionsFromPartSpec.java
+++ b/metastore/src/test/org/apache/hadoop/hive/metastore/client/TestAddPartitionsFromPartSpec.java
@@ -23,7 +23,6 @@
 import java.util.HashSet;
 import java.util.List;
 import java.util.Map;
-import java.util.Set;
 
 import org.apache.hadoop.fs.Path;
 import org.apache.hadoop.hive.metastore.IMetaStoreClient;
@@ -48,7 +47,6 @@
 import org.apache.thrift.TException;
 import org.apache.thrift.transport.TTransportException;
 import org.junit.After;
-import org.junit.AfterClass;
 import org.junit.Assert;
 import org.junit.Before;
 import org.junit.Test;
@@ -61,12 +59,7 @@
  * Tests for creating partitions from partition spec.
  */
 @RunWith(Parameterized.class)
-public class TestAddPartitionsFromPartSpec {
-
-  // Needed until there is no junit release with @BeforeParam, @AfterParam (junit 4.13)
-  // https://github.com/junit-team/junit4/commit/1bf8438b65858565dbb64736bfe13aae9cfc1b5a
-  // Then we should remove our own copy
-  private static Set<AbstractMetaStoreService> metaStoreServices = null;
+public class TestAddPartitionsFromPartSpec extends MetaStoreClientTest {
   private AbstractMetaStoreService metaStore;
   private IMetaStoreClient client;
 
@@ -81,29 +74,8 @@
   private static final int DEFAULT_CREATE_TIME = 123456;
   private static final short MAX = -1;
 
-  @Parameterized.Parameters(name = "{0}")
-  public static List<Object[]> getMetaStoreToTest() throws Exception {
-    metaStoreServices = new HashSet<AbstractMetaStoreService>();
-    List<Object[]> result = MetaStoreFactoryForTests.getMetaStores();
-    for(Object[] test: result) {
-      metaStoreServices.add((AbstractMetaStoreService)test[1]);
-    }
-    return result;
-  }
-  public TestAddPartitionsFromPartSpec(String name, AbstractMetaStoreService metaStore)
-      throws Exception {
+  public TestAddPartitionsFromPartSpec(String name, AbstractMetaStoreService metaStore) {
     this.metaStore = metaStore;
-    this.metaStore.start();
-  }
-
-  // Needed until there is no junit release with @BeforeParam, @AfterParam (junit 4.13)
-  // https://github.com/junit-team/junit4/commit/1bf8438b65858565dbb64736bfe13aae9cfc1b5a
-  // Then we should move this to @AfterParam
-  @AfterClass
-  public static void stopMetaStores() throws Exception {
-    for(AbstractMetaStoreService metaStoreService : metaStoreServices) {
-      metaStoreService.stop();
-    }
   }
 
   @Before
diff --git a/metastore/src/test/org/apache/hadoop/hive/metastore/client/TestAlterPartitions.java b/metastore/src/test/org/apache/hadoop/hive/metastore/client/TestAlterPartitions.java
index c82ca26..957f580 100644
--- a/metastore/src/test/org/apache/hadoop/hive/metastore/client/TestAlterPartitions.java
+++ b/metastore/src/test/org/apache/hadoop/hive/metastore/client/TestAlterPartitions.java
@@ -59,13 +59,8 @@
  * API tests for HMS client's  alterPartitions methods.
  */
 @RunWith(Parameterized.class)
-public class TestAlterPartitions {
-
+public class TestAlterPartitions extends MetaStoreClientTest {
   public static final int NEW_CREATE_TIME = 123456789;
-  // Needed until there is no junit release with @BeforeParam, @AfterParam (junit 4.13)
-  // https://github.com/junit-team/junit4/commit/1bf8438b65858565dbb64736bfe13aae9cfc1b5a
-  // Then we should remove our own copy
-  private static Set<AbstractMetaStoreService> metaStoreServices = null;
   private AbstractMetaStoreService metaStore;
   private IMetaStoreClient client;
 
@@ -73,30 +68,8 @@
   private static final String TABLE_NAME = "testparttable";
   private static final List<String> PARTCOL_SCHEMA = Lists.newArrayList("yyyy", "mm", "dd");
 
-
-  @Parameterized.Parameters(name = "{0}")
-  public static List<Object[]> getMetaStoreToTest() throws Exception {
-    metaStoreServices = new HashSet<AbstractMetaStoreService>();
-    List<Object[]> result = MetaStoreFactoryForTests.getMetaStores();
-    for(Object[] test: result) {
-      metaStoreServices.add((AbstractMetaStoreService)test[1]);
-    }
-    return result;
-  }
-
-  public TestAlterPartitions(String name, AbstractMetaStoreService metaStore) throws Exception {
+  public TestAlterPartitions(String name, AbstractMetaStoreService metaStore) {
     this.metaStore = metaStore;
-    this.metaStore.start();
-  }
-
-  // Needed until there is no junit release with @BeforeParam, @AfterParam (junit 4.13)
-  // https://github.com/junit-team/junit4/commit/1bf8438b65858565dbb64736bfe13aae9cfc1b5a
-  // Then we should move this to @AfterParam
-  @AfterClass
-  public static void stopMetaStores() throws Exception {
-    for(AbstractMetaStoreService metaStoreService : metaStoreServices) {
-      metaStoreService.stop();
-    }
   }
 
   @Before
diff --git a/metastore/src/test/org/apache/hadoop/hive/metastore/client/TestAppendPartitions.java b/metastore/src/test/org/apache/hadoop/hive/metastore/client/TestAppendPartitions.java
index 125bee1..27cf8ca 100644
--- a/metastore/src/test/org/apache/hadoop/hive/metastore/client/TestAppendPartitions.java
+++ b/metastore/src/test/org/apache/hadoop/hive/metastore/client/TestAppendPartitions.java
@@ -56,12 +56,7 @@
  * Tests for appending partitions.
  */
 @RunWith(Parameterized.class)
-public class TestAppendPartitions {
-
-  // Needed until there is no junit release with @BeforeParam, @AfterParam (junit 4.13)
-  // https://github.com/junit-team/junit4/commit/1bf8438b65858565dbb64736bfe13aae9cfc1b5a
-  // Then we should remove our own copy
-  private static Set<AbstractMetaStoreService> metaStoreServices = null;
+public class TestAppendPartitions extends MetaStoreClientTest {
   private AbstractMetaStoreService metaStore;
   private IMetaStoreClient client;
 
@@ -71,29 +66,8 @@
   private static Table tableNoPartColumns;
   private static Table tableView;
 
-  @Parameterized.Parameters(name = "{0}")
-  public static List<Object[]> getMetaStoreToTest() throws Exception {
-    metaStoreServices = new HashSet<AbstractMetaStoreService>();
-    List<Object[]> result = MetaStoreFactoryForTests.getMetaStores();
-    for(Object[] test: result) {
-      metaStoreServices.add((AbstractMetaStoreService)test[1]);
-    }
-    return result;
-  }
-
-  public TestAppendPartitions(String name, AbstractMetaStoreService metaStore) throws Exception {
+  public TestAppendPartitions(String name, AbstractMetaStoreService metaStore) {
     this.metaStore = metaStore;
-    this.metaStore.start();
-  }
-
-  // Needed until there is no junit release with @BeforeParam, @AfterParam (junit 4.13)
-  // https://github.com/junit-team/junit4/commit/1bf8438b65858565dbb64736bfe13aae9cfc1b5a
-  // Then we should move this to @AfterParam
-  @AfterClass
-  public static void stopMetaStores() throws Exception {
-    for (AbstractMetaStoreService metaStoreService : metaStoreServices) {
-      metaStoreService.stop();
-    }
   }
 
   @Before
diff --git a/metastore/src/test/org/apache/hadoop/hive/metastore/client/TestDatabases.java b/metastore/src/test/org/apache/hadoop/hive/metastore/client/TestDatabases.java
index 61c8e5b..8e7977f 100644
--- a/metastore/src/test/org/apache/hadoop/hive/metastore/client/TestDatabases.java
+++ b/metastore/src/test/org/apache/hadoop/hive/metastore/client/TestDatabases.java
@@ -38,61 +38,24 @@
 import org.apache.thrift.TException;
 import org.apache.thrift.transport.TTransportException;
 import org.junit.After;
-import org.junit.AfterClass;
 import org.junit.Assert;
 import org.junit.Before;
 import org.junit.Test;
 import org.junit.runner.RunWith;
 import org.junit.runners.Parameterized;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
 
 import java.util.HashMap;
-import java.util.HashSet;
 import java.util.List;
-import java.util.Set;
 
 @RunWith(Parameterized.class)
-public class TestDatabases {
-  private static final Logger LOG = LoggerFactory.getLogger(TestDatabases.class);
-  // Needed until there is no junit release with @BeforeParam, @AfterParam (junit 4.13)
-  // https://github.com/junit-team/junit4/commit/1bf8438b65858565dbb64736bfe13aae9cfc1b5a
-  // Then we should remove our own copy
-  private static Set<AbstractMetaStoreService> metaStoreServices = null;
+public class TestDatabases extends MetaStoreClientTest {
   private static final String DEFAULT_DATABASE = "default";
   private final AbstractMetaStoreService metaStore;
   private IMetaStoreClient client;
   private Database[] testDatabases = new Database[4];
 
-  @Parameterized.Parameters(name = "{0}")
-  public static List<Object[]> getMetaStoreToTest() throws Exception {
-    metaStoreServices = new HashSet<AbstractMetaStoreService>();
-    List<Object[]> result = MetaStoreFactoryForTests.getMetaStores();
-    for(Object[] test: result) {
-      metaStoreServices.add((AbstractMetaStoreService)test[1]);
-    }
-    return result;
-  }
-
-  public TestDatabases(String name, AbstractMetaStoreService metaStore) throws Exception {
+  public TestDatabases(String name, AbstractMetaStoreService metaStore) {
     this.metaStore = metaStore;
-    this.metaStore.start();
-  }
-
-  // Needed until there is no junit release with @BeforeParam, @AfterParam (junit 4.13)
-  // https://github.com/junit-team/junit4/commit/1bf8438b65858565dbb64736bfe13aae9cfc1b5a
-  // Then we should move this to @AfterParam
-  @AfterClass
-  public static void stopMetaStores() throws Exception {
-    for(AbstractMetaStoreService metaStoreService : metaStoreServices) {
-      try {
-        metaStoreService.stop();
-      } catch(Exception e) {
-        // Catch the exceptions, so every other metastore could be stopped as well
-        // Log it, so at least there is a slight possibility we find out about this :)
-        LOG.error("Error stopping MetaStoreService", e);
-      }
-    }
   }
 
   @Before
diff --git a/metastore/src/test/org/apache/hadoop/hive/metastore/client/TestDropPartitions.java b/metastore/src/test/org/apache/hadoop/hive/metastore/client/TestDropPartitions.java
index de2d4a3..85dc4c4 100644
--- a/metastore/src/test/org/apache/hadoop/hive/metastore/client/TestDropPartitions.java
+++ b/metastore/src/test/org/apache/hadoop/hive/metastore/client/TestDropPartitions.java
@@ -22,7 +22,6 @@
 import java.util.HashSet;
 import java.util.List;
 import java.util.Map;
-import java.util.Set;
 
 import org.apache.hadoop.fs.Path;
 import org.apache.hadoop.hive.conf.HiveConf;
@@ -39,9 +38,9 @@
 import org.apache.hadoop.hive.metastore.client.builder.TableBuilder;
 import org.apache.hadoop.hive.metastore.minihms.AbstractMetaStoreService;
 import org.junit.After;
-import org.junit.AfterClass;
 import org.junit.Assert;
 import org.junit.Before;
+import org.junit.BeforeClass;
 import org.junit.Test;
 import org.junit.runner.RunWith;
 import org.junit.runners.Parameterized;
@@ -52,12 +51,7 @@
  * Tests for dropping partitions.
  */
 @RunWith(Parameterized.class)
-public class TestDropPartitions {
-
-  // Needed until there is no junit release with @BeforeParam, @AfterParam (junit 4.13)
-  // https://github.com/junit-team/junit4/commit/1bf8438b65858565dbb64736bfe13aae9cfc1b5a
-  // Then we should remove our own copy
-  private static Set<AbstractMetaStoreService> metaStoreServices = null;
+public class TestDropPartitions extends MetaStoreClientTest {
   private AbstractMetaStoreService metaStore;
   private IMetaStoreClient client;
 
@@ -69,35 +63,19 @@
   private static final short MAX = -1;
   private static final Partition[] PARTITIONS = new Partition[3];
 
-  @Parameterized.Parameters(name = "{0}")
-  public static List<Object[]> getMetaStoreToTest() throws Exception {
-    metaStoreServices = new HashSet<AbstractMetaStoreService>();
-    List<Object[]> result = MetaStoreFactoryForTests.getMetaStores();
-    for(Object[] test: result) {
-      metaStoreServices.add((AbstractMetaStoreService)test[1]);
-    }
-    return result;
-  }
-
-  public TestDropPartitions(String name, AbstractMetaStoreService metaStore) throws Exception {
-    this.metaStore = metaStore;
+  @BeforeClass
+  public static void startMetaStores() {
     Map<HiveConf.ConfVars, String> msConf = new HashMap<HiveConf.ConfVars, String>();
     // Enable trash, so it can be tested
     Map<String, String> extraConf = new HashMap<String, String>();
     extraConf.put("fs.trash.checkpoint.interval", "30");  // FS_TRASH_CHECKPOINT_INTERVAL_KEY
     extraConf.put("fs.trash.interval", "30");             // FS_TRASH_INTERVAL_KEY (hadoop-2)
 
-    this.metaStore.start(msConf, extraConf);
+    startMetaStores(msConf, extraConf);
   }
 
-  // Needed until there is no junit release with @BeforeParam, @AfterParam (junit 4.13)
-  // https://github.com/junit-team/junit4/commit/1bf8438b65858565dbb64736bfe13aae9cfc1b5a
-  // Then we should move this to @AfterParam
-  @AfterClass
-  public static void stopMetaStores() throws Exception {
-    for(AbstractMetaStoreService metaStoreService : metaStoreServices) {
-      metaStoreService.stop();
-    }
+  public TestDropPartitions(String name, AbstractMetaStoreService metaStore) {
+    this.metaStore = metaStore;
   }
 
   @Before
diff --git a/metastore/src/test/org/apache/hadoop/hive/metastore/client/TestFunctions.java b/metastore/src/test/org/apache/hadoop/hive/metastore/client/TestFunctions.java
index d7eeb2a..2954b4b 100644
--- a/metastore/src/test/org/apache/hadoop/hive/metastore/client/TestFunctions.java
+++ b/metastore/src/test/org/apache/hadoop/hive/metastore/client/TestFunctions.java
@@ -35,7 +35,6 @@
 import org.apache.thrift.TApplicationException;
 import org.apache.thrift.transport.TTransportException;
 import org.junit.After;
-import org.junit.AfterClass;
 import org.junit.Assert;
 import org.junit.Before;
 import org.junit.Test;
@@ -44,17 +43,12 @@
 
 import java.util.HashSet;
 import java.util.List;
-import java.util.Set;
 
 /**
  * Test class for IMetaStoreClient API. Testing the Function related functions.
  */
 @RunWith(Parameterized.class)
-public class TestFunctions {
-  // Needed until there is no junit release with @BeforeParam, @AfterParam (junit 4.13)
-  // https://github.com/junit-team/junit4/commit/1bf8438b65858565dbb64736bfe13aae9cfc1b5a
-  // Then we should remove our own copy
-  private static Set<AbstractMetaStoreService> metaStoreServices = null;
+public class TestFunctions extends MetaStoreClientTest {
   private static final String DEFAULT_DATABASE = "default";
   private static final String OTHER_DATABASE = "dummy";
   private static final String TEST_FUNCTION_CLASS =
@@ -63,29 +57,8 @@
   private IMetaStoreClient client;
   private Function[] testFunctions = new Function[4];
 
-  @Parameterized.Parameters(name = "{0}")
-  public static List<Object[]> getMetaStoreToTest() throws Exception {
-    metaStoreServices = new HashSet<AbstractMetaStoreService>();
-    List<Object[]> result = MetaStoreFactoryForTests.getMetaStores();
-    for(Object[] test: result) {
-      metaStoreServices.add((AbstractMetaStoreService)test[1]);
-    }
-    return result;
-  }
-
-  public TestFunctions(String name, AbstractMetaStoreService metaStore) throws Exception {
+  public TestFunctions(String name, AbstractMetaStoreService metaStore) {
     this.metaStore = metaStore;
-    this.metaStore.start();
-  }
-
-  // Needed until there is no junit release with @BeforeParam, @AfterParam (junit 4.13)
-  // https://github.com/junit-team/junit4/commit/1bf8438b65858565dbb64736bfe13aae9cfc1b5a
-  // Then we should move this to @AfterParam
-  @AfterClass
-  public static void stopMetaStores() throws Exception {
-    for(AbstractMetaStoreService metaStoreService : metaStoreServices) {
-      metaStoreService.stop();
-    }
   }
 
   @Before
diff --git a/metastore/src/test/org/apache/hadoop/hive/metastore/client/TestGetListIndexes.java b/metastore/src/test/org/apache/hadoop/hive/metastore/client/TestGetListIndexes.java
index ca939d0..6283919 100644
--- a/metastore/src/test/org/apache/hadoop/hive/metastore/client/TestGetListIndexes.java
+++ b/metastore/src/test/org/apache/hadoop/hive/metastore/client/TestGetListIndexes.java
@@ -19,7 +19,6 @@
 
 import java.util.HashSet;
 import java.util.List;
-import java.util.Set;
 
 import org.apache.hadoop.hive.metastore.IMetaStoreClient;
 import org.apache.hadoop.hive.metastore.api.Database;
@@ -33,7 +32,6 @@
 import org.apache.hadoop.hive.metastore.minihms.AbstractMetaStoreService;
 import org.apache.thrift.TException;
 import org.junit.After;
-import org.junit.AfterClass;
 import org.junit.Assert;
 import org.junit.Before;
 import org.junit.Test;
@@ -46,11 +44,7 @@
  * Tests for getting and listing indexes.
  */
 @RunWith(Parameterized.class)
-public class TestGetListIndexes {
-  // Needed until there is no junit release with @BeforeParam, @AfterParam (junit 4.13)
-  // https://github.com/junit-team/junit4/commit/1bf8438b65858565dbb64736bfe13aae9cfc1b5a
-  // Then we should remove our own copy
-  private static Set<AbstractMetaStoreService> metaStoreServices = null;
+public class TestGetListIndexes extends MetaStoreClientTest {
   private AbstractMetaStoreService metaStore;
   private IMetaStoreClient client;
 
@@ -67,29 +61,8 @@
   private static final Index[] INDEXES = new Index[5];
   private static final short MAX = -1;
 
-  @Parameterized.Parameters(name = "{0}")
-  public static List<Object[]> getMetaStoreToTest() throws Exception {
-    metaStoreServices = new HashSet<AbstractMetaStoreService>();
-    List<Object[]> result = MetaStoreFactoryForTests.getMetaStores();
-    for(Object[] test: result) {
-      metaStoreServices.add((AbstractMetaStoreService)test[1]);
-    }
-    return result;
-  }
-
-  public TestGetListIndexes(String name, AbstractMetaStoreService metaStore) throws Exception {
+  public TestGetListIndexes(String name, AbstractMetaStoreService metaStore) {
     this.metaStore = metaStore;
-    this.metaStore.start();
-  }
-
-  // Needed until there is no junit release with @BeforeParam, @AfterParam (junit 4.13)
-  // https://github.com/junit-team/junit4/commit/1bf8438b65858565dbb64736bfe13aae9cfc1b5a
-  // Then we should move this to @AfterParam
-  @AfterClass
-  public static void stopMetaStores() throws Exception {
-    for (AbstractMetaStoreService metaStoreService : metaStoreServices) {
-      metaStoreService.stop();
-    }
   }
 
   @Before
diff --git a/metastore/src/test/org/apache/hadoop/hive/metastore/client/TestGetPartitions.java b/metastore/src/test/org/apache/hadoop/hive/metastore/client/TestGetPartitions.java
index f3360d2..7f9da71 100644
--- a/metastore/src/test/org/apache/hadoop/hive/metastore/client/TestGetPartitions.java
+++ b/metastore/src/test/org/apache/hadoop/hive/metastore/client/TestGetPartitions.java
@@ -20,7 +20,6 @@
 
 import java.util.HashSet;
 import java.util.List;
-import java.util.Set;
 
 import org.apache.hadoop.hive.metastore.IMetaStoreClient;
 import org.apache.hadoop.hive.metastore.api.Database;
@@ -38,7 +37,6 @@
 import com.google.common.collect.Lists;
 
 import org.junit.After;
-import org.junit.AfterClass;
 import org.junit.Before;
 import org.junit.Test;
 import org.junit.runner.RunWith;
@@ -53,42 +51,15 @@
  * API tests for HMS client's getPartitions methods.
  */
 @RunWith(Parameterized.class)
-public class TestGetPartitions {
-
-  // Needed until there is no junit release with @BeforeParam, @AfterParam (junit 4.13)
-  // https://github.com/junit-team/junit4/commit/1bf8438b65858565dbb64736bfe13aae9cfc1b5a
-  // Then we should remove our own copy
-  private static Set<AbstractMetaStoreService> metaStoreServices = null;
+public class TestGetPartitions extends MetaStoreClientTest {
   private AbstractMetaStoreService metaStore;
   private IMetaStoreClient client;
 
   private static final String DB_NAME = "testpartdb";
   private static final String TABLE_NAME = "testparttable";
 
-
-  @Parameterized.Parameters(name = "{0}")
-  public static List<Object[]> getMetaStoreToTest() throws Exception {
-    metaStoreServices = new HashSet<AbstractMetaStoreService>();
-    List<Object[]> result = MetaStoreFactoryForTests.getMetaStores();
-    for(Object[] test: result) {
-      metaStoreServices.add((AbstractMetaStoreService)test[1]);
-    }
-    return result;
-  }
-
-  public TestGetPartitions(String name, AbstractMetaStoreService metaStore) throws Exception {
+  public TestGetPartitions(String name, AbstractMetaStoreService metaStore) {
     this.metaStore = metaStore;
-    this.metaStore.start();
-  }
-
-  // Needed until there is no junit release with @BeforeParam, @AfterParam (junit 4.13)
-  // https://github.com/junit-team/junit4/commit/1bf8438b65858565dbb64736bfe13aae9cfc1b5a
-  // Then we should move this to @AfterParam
-  @AfterClass
-  public static void stopMetaStores() throws Exception {
-    for(AbstractMetaStoreService metaStoreService : metaStoreServices) {
-      metaStoreService.stop();
-    }
   }
 
   @Before
diff --git a/metastore/src/test/org/apache/hadoop/hive/metastore/client/TestGetTableMeta.java b/metastore/src/test/org/apache/hadoop/hive/metastore/client/TestGetTableMeta.java
index 94826f7..b8367c2 100644
--- a/metastore/src/test/org/apache/hadoop/hive/metastore/client/TestGetTableMeta.java
+++ b/metastore/src/test/org/apache/hadoop/hive/metastore/client/TestGetTableMeta.java
@@ -37,7 +37,6 @@
 import com.google.common.collect.Lists;
 
 import org.junit.After;
-import org.junit.AfterClass;
 import org.junit.Before;
 import org.junit.Test;
 import org.junit.runner.RunWith;
@@ -50,12 +49,7 @@
  * API tests for HMS client's getTableMeta method.
  */
 @RunWith(Parameterized.class)
-public class TestGetTableMeta {
-
-  // Needed until there is no junit release with @BeforeParam, @AfterParam (junit 4.13)
-  // https://github.com/junit-team/junit4/commit/1bf8438b65858565dbb64736bfe13aae9cfc1b5a
-  // Then we should remove our own copy
-  private static Set<AbstractMetaStoreService> metaStoreServices = null;
+public class TestGetTableMeta extends MetaStoreClientTest {
   private AbstractMetaStoreService metaStore;
   private IMetaStoreClient client;
 
@@ -64,30 +58,8 @@
 
   private List<TableMeta> expectedMetas = null;
 
-
-  @Parameterized.Parameters(name = "{0}")
-  public static List<Object[]> getMetaStoreToTest() throws Exception {
-    metaStoreServices = new HashSet<AbstractMetaStoreService>();
-    List<Object[]> result = MetaStoreFactoryForTests.getMetaStores();
-    for(Object[] test: result) {
-      metaStoreServices.add((AbstractMetaStoreService)test[1]);
-    }
-    return result;
-  }
-
-  public TestGetTableMeta(String name, AbstractMetaStoreService metaStore) throws Exception {
+  public TestGetTableMeta(String name, AbstractMetaStoreService metaStore) {
     this.metaStore = metaStore;
-    this.metaStore.start();
-  }
-
-  // Needed until there is no junit release with @BeforeParam, @AfterParam (junit 4.13)
-  // https://github.com/junit-team/junit4/commit/1bf8438b65858565dbb64736bfe13aae9cfc1b5a
-  // Then we should move this to @AfterParam
-  @AfterClass
-  public static void stopMetaStores() throws Exception {
-    for(AbstractMetaStoreService metaStoreService : metaStoreServices) {
-      metaStoreService.stop();
-    }
   }
 
   @Before
diff --git a/metastore/src/test/org/apache/hadoop/hive/metastore/client/TestListPartitions.java b/metastore/src/test/org/apache/hadoop/hive/metastore/client/TestListPartitions.java
index 4ff7c3a..7abbb26 100644
--- a/metastore/src/test/org/apache/hadoop/hive/metastore/client/TestListPartitions.java
+++ b/metastore/src/test/org/apache/hadoop/hive/metastore/client/TestListPartitions.java
@@ -21,12 +21,10 @@
 import java.util.ArrayList;
 import java.util.HashSet;
 import java.util.List;
-import java.util.Set;
 
 import org.apache.commons.lang.StringUtils;
 import org.apache.hadoop.hive.metastore.IMetaStoreClient;
 import org.apache.hadoop.hive.metastore.api.Database;
-import org.apache.hadoop.hive.metastore.api.FieldSchema;
 import org.apache.hadoop.hive.metastore.api.MetaException;
 import org.apache.hadoop.hive.metastore.api.NoSuchObjectException;
 import org.apache.hadoop.hive.metastore.api.Partition;
@@ -38,13 +36,11 @@
 import org.apache.hadoop.hive.metastore.minihms.AbstractMetaStoreService;
 import org.apache.hadoop.hive.metastore.partition.spec.PartitionSpecProxy;
 import org.apache.thrift.TException;
-import org.apache.thrift.protocol.TProtocolException;
 import org.apache.thrift.transport.TTransportException;
 
 import com.google.common.collect.Lists;
 
 import org.junit.After;
-import org.junit.AfterClass;
 import org.junit.Before;
 import org.junit.Test;
 import org.junit.runner.RunWith;
@@ -61,41 +57,15 @@
  * API tests for HMS client's listPartitions methods.
  */
 @RunWith(Parameterized.class)
-public class TestListPartitions {
-
-  // Needed until there is no junit release with @BeforeParam, @AfterParam (junit 4.13)
-  // https://github.com/junit-team/junit4/commit/1bf8438b65858565dbb64736bfe13aae9cfc1b5a
-  // Then we should remove our own copy
-  private static Set<AbstractMetaStoreService> metaStoreServices = null;
+public class TestListPartitions extends MetaStoreClientTest {
   private AbstractMetaStoreService metaStore;
   private IMetaStoreClient client;
 
   private static final String DB_NAME = "testpartdb";
   private static final String TABLE_NAME = "testparttable";
 
-  @Parameterized.Parameters(name = "{0}")
-  public static List<Object[]> getMetaStoreToTest() throws Exception {
-    metaStoreServices = new HashSet<AbstractMetaStoreService>();
-    List<Object[]> result = MetaStoreFactoryForTests.getMetaStores();
-    for(Object[] test: result) {
-      metaStoreServices.add((AbstractMetaStoreService)test[1]);
-    }
-    return result;
-  }
-
-  public TestListPartitions(String name, AbstractMetaStoreService metaStore) throws Exception {
+  public TestListPartitions(String name, AbstractMetaStoreService metaStore) {
     this.metaStore = metaStore;
-    this.metaStore.start();
-  }
-
-  // Needed until there is no junit release with @BeforeParam, @AfterParam (junit 4.13)
-  // https://github.com/junit-team/junit4/commit/1bf8438b65858565dbb64736bfe13aae9cfc1b5a
-  // Then we should move this to @AfterParam
-  @AfterClass
-  public static void stopMetaStores() throws Exception {
-    for(AbstractMetaStoreService metaStoreService : metaStoreServices) {
-      metaStoreService.stop();
-    }
   }
 
   @Before
diff --git a/metastore/src/test/org/apache/hadoop/hive/metastore/client/TestTablesCreateDropAlterTruncate.java b/metastore/src/test/org/apache/hadoop/hive/metastore/client/TestTablesCreateDropAlterTruncate.java
index e5c876a..bd00fbf 100644
--- a/metastore/src/test/org/apache/hadoop/hive/metastore/client/TestTablesCreateDropAlterTruncate.java
+++ b/metastore/src/test/org/apache/hadoop/hive/metastore/client/TestTablesCreateDropAlterTruncate.java
@@ -41,9 +41,9 @@
 import org.apache.thrift.protocol.TProtocolException;
 import org.apache.thrift.transport.TTransportException;
 import org.junit.After;
-import org.junit.AfterClass;
 import org.junit.Assert;
 import org.junit.Before;
+import org.junit.BeforeClass;
 import org.junit.Test;
 import org.junit.runner.RunWith;
 import org.junit.runners.Parameterized;
@@ -53,18 +53,13 @@
 import java.util.HashSet;
 import java.util.List;
 import java.util.Map;
-import java.util.Set;
 
 /**
  * Test class for IMetaStoreClient API. Testing the Table related functions for metadata
  * manipulation, like creating, dropping and altering tables.
  */
 @RunWith(Parameterized.class)
-public class TestTablesCreateDropAlterTruncate {
-  // Needed until there is no junit release with @BeforeParam, @AfterParam (junit 4.13)
-  // https://github.com/junit-team/junit4/commit/1bf8438b65858565dbb64736bfe13aae9cfc1b5a
-  // Then we should remove our own copy
-  private static Set<AbstractMetaStoreService> metaStoreServices = null;
+public class TestTablesCreateDropAlterTruncate extends MetaStoreClientTest {
   private static final String DEFAULT_DATABASE = "default";
   private static final String OTHER_DATABASE = "dummy";
   private final AbstractMetaStoreService metaStore;
@@ -73,36 +68,19 @@
   private Table partitionedTable = null;
   private Table externalTable = null;
 
-
-  @Parameterized.Parameters(name = "{0}")
-  public static List<Object[]> getMetaStoreToTest() throws Exception {
-    metaStoreServices = new HashSet<AbstractMetaStoreService>();
-    List<Object[]> result = MetaStoreFactoryForTests.getMetaStores();
-    for(Object[] test: result) {
-      metaStoreServices.add((AbstractMetaStoreService)test[1]);
-    }
-    return result;
+  public TestTablesCreateDropAlterTruncate(String name, AbstractMetaStoreService metaStore) {
+    this.metaStore = metaStore;
   }
 
-  public TestTablesCreateDropAlterTruncate(String name, AbstractMetaStoreService metaStore) throws Exception {
-    this.metaStore = metaStore;
+  @BeforeClass
+  public static void startMetaStores() {
     Map<HiveConf.ConfVars, String> msConf = new HashMap<HiveConf.ConfVars, String>();
     // Enable trash, so it can be tested
     Map<String, String> extraConf = new HashMap<String, String>();
     extraConf.put("fs.trash.checkpoint.interval", "30");  // FS_TRASH_CHECKPOINT_INTERVAL_KEY
     extraConf.put("fs.trash.interval", "30");             // FS_TRASH_INTERVAL_KEY (hadoop-2)
 
-    this.metaStore.start(msConf, extraConf);
-  }
-
-  // Needed until there is no junit release with @BeforeParam, @AfterParam (junit 4.13)
-  // https://github.com/junit-team/junit4/commit/1bf8438b65858565dbb64736bfe13aae9cfc1b5a
-  // Then we should move this to @AfterParam
-  @AfterClass
-  public static void stopMetaStores() throws Exception {
-    for(AbstractMetaStoreService metaStoreService : metaStoreServices) {
-      metaStoreService.stop();
-    }
+    startMetaStores(msConf, extraConf);
   }
 
   @Before
diff --git a/metastore/src/test/org/apache/hadoop/hive/metastore/client/TestTablesGetExists.java b/metastore/src/test/org/apache/hadoop/hive/metastore/client/TestTablesGetExists.java
index 5ffa244..bfee7eb 100644
--- a/metastore/src/test/org/apache/hadoop/hive/metastore/client/TestTablesGetExists.java
+++ b/metastore/src/test/org/apache/hadoop/hive/metastore/client/TestTablesGetExists.java
@@ -24,14 +24,12 @@
 import org.apache.hadoop.hive.metastore.api.NoSuchObjectException;
 import org.apache.hadoop.hive.metastore.api.Table;
 import org.apache.hadoop.hive.metastore.api.UnknownDBException;
-import org.apache.hadoop.hive.metastore.api.hive_metastoreConstants;
 import org.apache.hadoop.hive.metastore.client.builder.DatabaseBuilder;
 import org.apache.hadoop.hive.metastore.client.builder.TableBuilder;
 import org.apache.hadoop.hive.metastore.minihms.AbstractMetaStoreService;
 import org.apache.thrift.protocol.TProtocolException;
 import org.apache.thrift.transport.TTransportException;
 import org.junit.After;
-import org.junit.AfterClass;
 import org.junit.Assert;
 import org.junit.Before;
 import org.junit.Test;
@@ -41,47 +39,21 @@
 import java.util.ArrayList;
 import java.util.HashSet;
 import java.util.List;
-import java.util.Set;
 
 /**
  * Test class for IMetaStoreClient API. Testing the Table related functions for metadata
  * querying like getting one, or multiple tables, and table name lists.
  */
 @RunWith(Parameterized.class)
-public class TestTablesGetExists {
-  // Needed until there is no junit release with @BeforeParam, @AfterParam (junit 4.13)
-  // https://github.com/junit-team/junit4/commit/1bf8438b65858565dbb64736bfe13aae9cfc1b5a
-  // Then we should remove our own copy
-  private static Set<AbstractMetaStoreService> metaStoreServices = null;
+public class TestTablesGetExists extends MetaStoreClientTest {
   private static final String DEFAULT_DATABASE = "default";
   private static final String OTHER_DATABASE = "dummy";
   private final AbstractMetaStoreService metaStore;
   private IMetaStoreClient client;
   private Table[] testTables = new Table[7];
 
-  @Parameterized.Parameters(name = "{0}")
-  public static List<Object[]> getMetaStoreToTest() throws Exception {
-    metaStoreServices = new HashSet<AbstractMetaStoreService>();
-    List<Object[]> result = MetaStoreFactoryForTests.getMetaStores();
-    for(Object[] test: result) {
-      metaStoreServices.add((AbstractMetaStoreService)test[1]);
-    }
-    return result;
-  }
-
-  public TestTablesGetExists(String name, AbstractMetaStoreService metaStore) throws Exception {
+  public TestTablesGetExists(String name, AbstractMetaStoreService metaStore) {
     this.metaStore = metaStore;
-    this.metaStore.start();
-  }
-
-  // Needed until there is no junit release with @BeforeParam, @AfterParam (junit 4.13)
-  // https://github.com/junit-team/junit4/commit/1bf8438b65858565dbb64736bfe13aae9cfc1b5a
-  // Then we should move this to @AfterParam
-  @AfterClass
-  public static void stopMetaStores() throws Exception {
-    for(AbstractMetaStoreService metaStoreService : metaStoreServices) {
-      metaStoreService.stop();
-    }
   }
 
   @Before
diff --git a/metastore/src/test/org/apache/hadoop/hive/metastore/client/TestTablesList.java b/metastore/src/test/org/apache/hadoop/hive/metastore/client/TestTablesList.java
index cf9b5ee..ff89898 100644
--- a/metastore/src/test/org/apache/hadoop/hive/metastore/client/TestTablesList.java
+++ b/metastore/src/test/org/apache/hadoop/hive/metastore/client/TestTablesList.java
@@ -21,67 +21,34 @@
 import org.apache.hadoop.hive.metastore.IMetaStoreClient;
 import org.apache.hadoop.hive.metastore.api.InvalidOperationException;
 import org.apache.hadoop.hive.metastore.api.MetaException;
-import org.apache.hadoop.hive.metastore.api.NoSuchObjectException;
 import org.apache.hadoop.hive.metastore.api.Table;
 import org.apache.hadoop.hive.metastore.api.UnknownDBException;
 import org.apache.hadoop.hive.metastore.api.hive_metastoreConstants;
 import org.apache.hadoop.hive.metastore.client.builder.DatabaseBuilder;
 import org.apache.hadoop.hive.metastore.client.builder.TableBuilder;
 import org.apache.hadoop.hive.metastore.minihms.AbstractMetaStoreService;
-import org.apache.thrift.protocol.TProtocolException;
-import org.apache.thrift.transport.TTransportException;
 import org.junit.After;
-import org.junit.AfterClass;
 import org.junit.Assert;
 import org.junit.Before;
 import org.junit.Test;
 import org.junit.runner.RunWith;
 import org.junit.runners.Parameterized;
-
-import java.util.ArrayList;
-import java.util.HashSet;
 import java.util.List;
-import java.util.Set;
 
 /**
  * Test class for IMetaStoreClient API. Testing the Table related functions for metadata
  * querying like getting one, or multiple tables, and table name lists.
  */
 @RunWith(Parameterized.class)
-public class TestTablesList {
-  // Needed until there is no junit release with @BeforeParam, @AfterParam (junit 4.13)
-  // https://github.com/junit-team/junit4/commit/1bf8438b65858565dbb64736bfe13aae9cfc1b5a
-  // Then we should remove our own copy
-  private static Set<AbstractMetaStoreService> metaStoreServices = null;
+public class TestTablesList extends MetaStoreClientTest {
   private static final String DEFAULT_DATABASE = "default";
   private static final String OTHER_DATABASE = "dummy";
   private final AbstractMetaStoreService metaStore;
   private IMetaStoreClient client;
   private Table[] testTables = new Table[7];
 
-  @Parameterized.Parameters(name = "{0}")
-  public static List<Object[]> getMetaStoreToTest() throws Exception {
-    metaStoreServices = new HashSet<AbstractMetaStoreService>();
-    List<Object[]> result = MetaStoreFactoryForTests.getMetaStores();
-    for(Object[] test: result) {
-      metaStoreServices.add((AbstractMetaStoreService)test[1]);
-    }
-    return result;
-  }
-
-  public TestTablesList(String name, AbstractMetaStoreService metaStore) throws Exception {
+  public TestTablesList(String name, AbstractMetaStoreService metaStore) {
     this.metaStore = metaStore;
-    this.metaStore.start();
-  }
-
-  // Needed until there is no junit release with @BeforeParam, @AfterParam (junit 4.13)
-  // https://github.com/junit-team/junit4/commit/1bf8438b65858565dbb64736bfe13aae9cfc1b5a
-  // Then we should move this to @AfterParam
-  @AfterClass
-  public static void stopMetaStores() throws Exception {
-    for(AbstractMetaStoreService metaStoreService : metaStoreServices) {
-      metaStoreService.stop();
-    }
   }
 
   @Before
-- 
1.7.9.5

